storage/filesystem/littlefs/source/LittleFileSystem.cpp

```diff
@@ -117,21 +123,55 @@ static int lfs_totype(int type)
 static int lfs_bd_read(const struct lfs_config *c, lfs_block_t block,
                        lfs_off_t off, void *buffer, lfs_size_t size)
 {
+#if COMPONENT_SPINAND
+#if NFTL
+    return nftl_flash_read(NFTL_PARTITION0, block, off, buffer, size);
```

Comment:
This won't work if we have both a SPINand and another flash, like an external QSPI flash (QSPIF), present in the system as the same time. Instead of `#if`, we should make use of the appropriate virtual methods on the BlockDevice instance.